dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/internal/utils/MiscUtilsTest.kt

```diff
@@ -90,40 +97,58 @@ internal class MiscUtilsTest {
 
     @Test
     fun `M do nothing W retryWithDelay { times less or equal than 0 }`(forge: Forge) {
-        // GIVEN
+        // Given
         val fakeDelay = TimeUnit.SECONDS.toNanos(forge.aLong(min = 0, max = 2))
         val mockedBlock: () -> Boolean = mock()
 
-        // WHEN
+        // When
         retryWithDelay(mockedBlock, forge.anInt(Int.MIN_VALUE, 1), fakeDelay, mockInternalLogger)
 
-        // THEN
+        // Then
         verifyNoInteractions(mockedBlock)
     }
 
     @Test
-    fun `M repeat until success W retryWithDelay`(forge: Forge) {
-        // GIVEN
+    fun `M repeat until success W retryWithDelay { result false }`(forge: Forge) {
+        // Given
         val fakeDelay = TimeUnit.SECONDS.toNanos(forge.aLong(min = 0, max = 2))
         val mockedBlock: () -> Boolean = mock()
         whenever(mockedBlock.invoke()).thenReturn(false).thenReturn(true)
 
-        // WHEN
+        // When
+        val wasSuccessful = retryWithDelay(mockedBlock, 3, fakeDelay, mockInternalLogger)
+
+        // Then
+        assertThat(wasSuccessful).isTrue()
+        verify(mockedBlock, times(2)).invoke()
+    }
+
+    @Test
+    fun `M repeat until success W retryWithDelay { exception }`(
+        @Forgery exception: Exception,
+        forge: Forge
+    ) {
+        // Given
+        val fakeDelay = TimeUnit.SECONDS.toNanos(forge.aLong(min = 0, max = 2))
+        val mockedBlock: () -> Boolean = mock()
+        whenever(mockedBlock.invoke()).thenAnswer { throw exception }.thenReturn(true)
```

Comment:
probably can be shorter a bit

```suggestion
        whenever(mockedBlock.invoke()).thenThrow(exception).thenReturn(true)
```

Reply:
Indeed 👍